### ISSUE TITLE
Incompatibility with werkzeug >= 2.2.0

### ISSUE THREAD

**Issue by frantisekz** [Flask-Middleware/flask-security#641]
It seems the testsuite of flask-security 4.1.4 isn't compatible with werkzeug >= 2.2 with following failures:

```
______________________________ test_basic_change _______________________________

app = <Flask 'tests.conftest'>
client_nc = <FlaskClient <Flask 'tests.conftest'>>
get_message = <function get_message.<locals>.fn at 0x7f51f4d042c0>

    @pytest.mark.settings(api_enabled_methods=["basic"])
    def test_basic_change(app, client_nc, get_message):
        # Verify can change password using basic auth
        data = dict(
            password="password",
            new_password="new strong password",
            new_password_confirm="new strong password",
        )
        response = client_nc.post("/change", data=data)
        assert b"You are not authenticated" in response.data
        assert "WWW-Authenticate" in response.headers
    
>       response = client_nc.post(
            "/change",
            data=data,
            headers={
                "Authorization": "Basic %s"
                % base64.b64encode(b"matt@lp.com:password").decode("utf-8")
            },
            follow_redirects=True,
        )

tests/test_changeable.py:429: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.11/site-packages/werkzeug/test.py:1140: in post
    return self.open(*args, **kw)
/usr/lib/python3.11/site-packages/flask/testing.py:217: in open
    return super().open(
/usr/lib/python3.11/site-packages/werkzeug/test.py:1089: in open
    response = self.run_wsgi_app(request.environ, buffered=buffered)
/usr/lib/python3.11/site-packages/werkzeug/test.py:956: in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
/usr/lib/python3.11/site-packages/werkzeug/test.py:1237: in run_wsgi_app
    app_rv = app(environ, start_response)
/usr/lib/python3.11/site-packages/flask/app.py:2091: in __call__
    return self.wsgi_app(environ, start_response)
/usr/lib/python3.11/site-packages/flask/app.py:2076: in wsgi_app
    response = self.handle_exception(e)
/usr/lib/python3.11/site-packages/flask/app.py:2073: in wsgi_app
    response = self.full_dispatch_request()
/usr/lib/python3.11/site-packages/flask/app.py:1519: in full_dispatch_request
    rv = self.handle_user_exception(e)
/usr/lib/python3.11/site-packages/flask/app.py:1517: in full_dispatch_request
    rv = self.dispatch_request()
/usr/lib/python3.11/site-packages/flask/app.py:1503: in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
flask_security/decorators.py:390: in decorated_view
    return fn(*args, **kwargs)
flask_security/views.py:641: in change_password
    if form.validate_on_submit():
/usr/lib/python3.11/site-packages/flask_wtf/form.py:86: in validate_on_submit
    return self.is_submitted() and self.validate()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <flask_security.forms.ChangePasswordForm object at 0x7f51f4be2e50>

    def validate(self):
        if not super().validate():
            return False
    
        self.password.data = _security._password_util.normalize(self.password.data)
>       if not current_user.verify_and_update_password(self.password.data):
E       AttributeError: 'AnonymousUser' object has no attribute 'verify_and_update_password'

flask_security/forms.py:606: AttributeError
_________________________ test_change_token_uniquifier _________________________

app = <Flask 'tests.conftest'>

    def test_change_token_uniquifier(app):
        # make sure that existing token no longer works once we change the token uniquifier
        from sqlalchemy import Column, String
        from flask_sqlalchemy import SQLAlchemy
        from flask_security.models import fsqla_v2 as fsqla
        from flask_security import Security, SQLAlchemyUserDatastore
    
        app.config["SQLALCHEMY_DATABASE_URI"] = "sqlite:///:memory:"
        db = SQLAlchemy(app)
    
        fsqla.FsModels.set_db_info(db)
    
        class Role(db.Model, fsqla.FsRoleMixin):
            pass
    
        class User(db.Model, fsqla.FsUserMixin):
            fs_token_uniquifier = Column(String(64), unique=True, nullable=False)
    
        with app.app_context():
            db.create_all()
    
        ds = SQLAlchemyUserDatastore(db, User, Role)
        app.security = Security(app, datastore=ds)
    
        with app.app_context():
            ds.create_user(
                email="matt@lp.com",
                password=hash_password("password"),
            )
            ds.commit()
    
            client_nc = app.test_client(use_cookies=False)
    
            response = json_authenticate(client_nc)
            token = response.json["response"]["user"]["authentication_token"]
            verify_token(client_nc, token)
    
            # now change uniquifier
            with app.test_request_context("/"):
                user = app.security.datastore.find_user(email="matt@lp.com")
                app.security.datastore.reset_user_access(user)
                app.security.datastore.commit()
    
            verify_token(client_nc, token, status=401)
    
            # get new token and verify it works
            response = json_authenticate(client_nc)
>           token = response.json["response"]["user"]["authentication_token"]
E           KeyError: 'user'

tests/test_common.py:831: KeyError
_______________________________ test_token_query _______________________________

in_app_context = <Flask 'tests.conftest'>

    def test_token_query(in_app_context):
        # Verify that when authenticating with auth token (and not session)
        # that there is just one DB query to get user.
        app = in_app_context
        populate_data(app)
        client_nc = app.test_client(use_cookies=False)
    
        response = json_authenticate(client_nc)
        token = response.json["response"]["user"]["authentication_token"]
        current_nqueries = get_num_queries(app.security.datastore)
    
        response = client_nc.get(
            "/token",
            headers={"Content-Type": "application/json", "Authentication-Token": token},
        )
        assert response.status_code == 200
        end_nqueries = get_num_queries(app.security.datastore)
>       assert current_nqueries is None or end_nqueries == (current_nqueries + 1)
E       assert (1 is None or 3 == (1 + 1))

tests/test_common.py:896: AssertionError
```

**Comment by jwag956** [comment]
Yup - working on it.

**Comment by jwag956** [comment]
Fixed in both a patch release 4.1.5 and in master.
